uwblueprint/sistering
# Schedule Review Table Data Fetch

## Ticket link

Closes #162


## Implementation description
* Created new endpoint `shiftsWithSignupsAndVolunteersByPosting`
* This is a draft PR, still have to add the front end data-fetching 



## Steps to test
1. Populate database as in this #90 
2. Add skills and branches for the user with the following queries: `
  * `insert into skills values (1, "test skill");`
  * `insert into branches values (1, "test branch");`
  * `insert into "_SkillToVolunteer" values (1, 1);`
  * `insert into  "_BranchToVolunteer" values (1, 1);`
3. Test the query to see if it works as expected in `:5000/graphql`
4. Go to frontend route `/admin/scheduleTableDemo` to see demo of query
5. Input appropriate posting ID, user ID, and signup status for testing

Note that the queries `createShifts`, `createShiftSignups`, and `updateShiftSignup` will likely be useful for creating shifts and signing up to them.


## What should reviewers focus on?
* The reviewer should focus on checking to see if the table is populated as expected with the different filters for posting ID, user ID, and signup status.


## Checklist
